### ISSUE TITLE
su: user EF6E286DDA85EA2A4BA7DE684E2C6E8793298290 does not exist

### ISSUE THREAD

**Issue by suedi** [oshazard/apacman#22]
Have a problem with lines 657  and 659
```bash
657: su -c "addpgpkeys ${validpgpkeys[@]}" aurbuild

659: su -c "removepgpkeys ${validpgpkeys[@]}" aurbuild
```
I have two keys in validpgpkeys

8738A680B84B3031A630F2DB416F061063FEE659
EF6E286DDA85EA2A4BA7DE684E2C6E8793298290

when trying to build tor-browser-en bash/su interprets the second key in validgpgkeys as a user

and I get the error message above

I don't know how to tell bash to parse in right order here

I did as a workaround 

```bash
tmp_validpgpkeys="${validpgpkeys[@]}"
su -c "addpgpkeys $tmp_validpgpkeys" aurbuild
...
 su -c "removepgpkeys  $tmp_validpgpkeys aurbuild
unset tmp_validpgpkeys
```

How to do this better

**Comment by oshazard** [comment]
Wow, hmmm. Thanks, I'll get right on this.

**Comment by oshazard** [comment]
There were several issues but finally got it **fixed for v2.1** (@022c404) !

The default behavior when presented with a signed package is to prompt, then import the keys, build the package and finally remove the PGP keys.

Alternatively, use the *--keepkeys* flag to avoid the removal. This can be added to *apacman.conf* as `keepflags=1`. Also provided is *--purgekeys* flag to remove pre-existing keys.

---------

For future reference: 
* when passing arrays to `su` use `${var[*]}` rather than `${var[@]}` (but otherwise simply double-quote "${var[@]}" to avoid expansion)
* pass username to `su` before `-c`
* separate user input from `su` (otherwise subshell `exit` may not work as intended)
